src/modules/queue-visualizer/QueueVisualizer.test.tsx

```diff
@@ -0,0 +1,61 @@
+import { describe, test } from "vitest";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import { QueueVisualizer } from "./QueueVisualizer";
+import { Provider } from "react-redux";
+import { store } from "../core/store/store";
+import { BrowserRouter } from "react-router-dom";
+
+describe("queueVisualizer test", () => {
+  test("Should render component ", () => {
+    render(
+      <Provider store={store}>
+        <QueueVisualizer />
+      </Provider>,
+      { wrapper: BrowserRouter }
+    );
+  });
+
+  test("Should have back button", () => {
+    render(
+      <Provider store={store}>
+        <QueueVisualizer />
+      </Provider>,
+      { wrapper: BrowserRouter }
+    );
+
+    expect(screen.getByTestId("data-list-back-button")).toBeInTheDocument();
+  });
+
+  test("Should process input and add button", () => {
+    render(
+      <Provider store={store}>
+        <QueueVisualizer />
+      </Provider>,
+      { wrapper: BrowserRouter }
+    );
+
+    const textInput = screen.getByTestId("data-list-text-input");
+    fireEvent.change(textInput, { target: { value: "queue elem" } });
+
+    const addButton = screen.getByTestId("data-list-add-button");
+    fireEvent.click(addButton);
+
+    expect(screen.getByText(/queue elem/i)).toBeInTheDocument();
```

Comment:
You are parsing the whole screen to find the value in the specific place. But you can add a data-testid for the block where it should be presented an find it there.